rust_src/src/buffers.rs

```diff
@@ -211,3 +213,30 @@ fn buffer_modified_tick(buffer: LispObject) -> LispObject {
 fn buffer_chars_modified_tick(buffer: LispObject) -> LispObject {
     LispObject::from_fixnum(buffer.as_buffer_or_current_buffer().chars_modiff())
 }
+
+#[no_mangle]
+pub extern "C" fn validate_region(b: *mut Lisp_Object, e: *mut Lisp_Object) {
+    let start = LispObject::from_raw(unsafe { *b });
+    let stop = LispObject::from_raw(unsafe { *e });
+
+    let mut beg = start.as_fixnum_coerce_marker_or_error();
+    let mut end = stop.as_fixnum_coerce_marker_or_error();
+
+    if beg > end {
+        mem::swap(&mut beg, &mut end);
+    }
+
+    unsafe {
+        *b = LispObject::from_fixnum(beg).to_raw();
+        *e = LispObject::from_fixnum(end).to_raw();
+    }
+
+    let buffer = current_buffer();
+    let buf = buffer.as_buffer_or_error();
```

Comment:
I need the current buffer as a LispObject for the `args_out_of_range!` call, but I can change it if you want and just call `current_buffer` there.